baremaps-iploc/src/main/java/com/baremaps/iploc/IpLoc.java

```diff
@@ -51,18 +61,30 @@ public IpLoc(String databaseUrl, Geocoder geocoder) {
    * @param nicObjects the stream of nic objects to import
    */
   public void insertNicObjects(Stream<NicObject> nicObjects) {
-    nicObjects.forEach(
-        nicObject -> {
-          if (nicObject.attributes().size() > 0) {
-            NicAttribute firstAttribute = nicObject.attributes().get(0);
-            if (Objects.equals(firstAttribute.name(), "inetnum")) {
-              try {
-                processInetnum(nicObject, firstAttribute);
-              } catch (Exception e) {
-                throw new RuntimeException(e);
-              }
-            }
-          }
+    Stream<NicObject> filteredNicObjects =
+        nicObjects.filter(
+            nicObject ->
+                nicObject.attributes().size() > 0
```

Comment:
What about extracting some of these nested lambdas in private or protected methods to make the code easier to read?